### ISSUE TITLE
include description

### ISSUE THREAD

**Issue by jardakotesovec** [GertjanReynaert/react-intl-translations-manager#4]
Very nice tool! 

I am wondering what do you provide to translators in your workflow.  It make sense to provide them also description from messages, not just id and original text, but I don't see such option.

**Comment by GertjanReynaert** [comment]
The description is definitely important to translators. That's basically what the original Babel plugins purpose is: extract all the messages, including their descriptions, so they can be easily passed to translators.

So I see 2 options: or you pass all the separate extracted message files to your translators, or you use this tool its `singleMessagesFile` to generate a single file from all the extracted messages and you pass that file to your translators.

The descriptions only need to be included with the original messages, and that's why the translation files themselves only contain the translations in a key value format.

This way your translators can use the description and the original message to determine accurate translations, and they can return you the translations in the expected key value format.

I hope this helps.

Small disclaimer: I haven't made use of a translators services so if any of the above I assumed doesn't make sense, please enlighten me :)

**Comment by GertjanReynaert** [comment]
I'll close this issue for now, if you have some more questions, please reopen

**Comment by jardakotesovec** [comment]
> This way your translators can use the description and the original message to determine accurate translations, and they can return you the translations in the expected key value format.

That would assume that translator has tool that can import this single file and this tool would use previously translated keys and export key:value. I thought that point of this manager is to manage whole workflow not just generate single file for further processing.

Still wondering about your translation workflow. It seems to me that logic is that locale file  (key:value json file) should be edited by translator. Then its easy to manage, as manager informs about removed/new/untranslated keys. Which is fine.. just would be handy to be able to get inline context - originalMessage and description. 

So basically I would expect that two files get generated - final *key:value json* and one which is supposed to be edited and contains: id, originalMessage, translation and description. Later file would be used as source for next iteration, so key:value json would get sync with recent translations.

Hope it makes sense.

**Comment by jardakotesovec** [comment]
@GertjanReynaert Fyi github does not let me to reopen since you closed it.

**Comment by GertjanReynaert** [comment]
The point of this translation manager is to tell you what messages are removed/added/untranslated from your different translation files so that managing your translations becomes easier, but it doesn't dictate or guide you in a way to add/update your translations.

> It seems to me that logic is that locale file (key:value json file) should be edited by translator.

This is indeed my current workflow: I edit these locale files myself. 

I get what you want to be generated as files, but I don't think it's a good idea. Currently you have your key:value translation files, which only contain your translations, And then you have the translation information which is located or in your components themselves, or in the babel plugin extracted messages, or in the generated single file. This way the editable parts are in the translation files, and the information/reference parts are located in a single place for lookup if needed.

I'm not planning on changing the file output of the default translation manager, but you can always use the [translation manager core](https://github.com/GertjanReynaert/react-intl-translations-manager#core) for creating this kind of behaviour. If you wan't some inspiration you can always look into how the [default manager](https://github.com/GertjanReynaert/react-intl-translations-manager/blob/master/src/manageTranslations.js) is built

**Comment by jardakotesovec** [comment]
@GertjanReynaert Ok, thanks for feedback!
